### ISSUE TITLE
Update eclipse jdt formatter configuration

### ISSUE THREAD

**Issue by marko-radosavljevic** [Together-Java/TJ-Bot#605]
We should update our formatting configuration. It's based on [old google style](https://github.com/google/styleguide/blob/gh-pages/eclipse-java-google-style.xml), and it lacks all the new features.

It also has a weird issue with formatting a specific record. Issue noticed first here: https://github.com/Together-Java/TJ-Bot/pull/140. Investigated by spotless devs here: https://github.com/diffplug/spotless/issues/951

![image](https://user-images.githubusercontent.com/30927961/193826907-69c07674-71b5-4e00-8ec9-25ebe6c25ee4.png)

After investigation, the conclusion is that our configuration is the issue. And our band-aid solution was to pin the eclipse jdt formatter version to the latest known working, `eclipse('4.19.0')`.

Partly because of that pinning, we don't have `@snippet` support. https://github.com/Together-Java/TJ-Bot/issues/591. And we will continue experiencing more and more issues down the line. 

* * *

It's also a good opportunity to rethink our formatter choice, use better and smarter configuration etc.

**Comment by Taz03** [comment]
i'd like to work on this. Should the configuration should be same as before?

**Comment by marko-radosavljevic** [comment]
Yeah, ideally it would be the same, or almost the same. It's not a hard requirement tho, we could agree on a different style, in case it's superior for whatever reason.

The best way we found to edit and test the configuration is in eclipse preferences, formatter edit tool. You can export or import configurations, and edit them in real time with a preview. 

Looks like this:
[editing eclipse formatter](https://user-images.githubusercontent.com/30927961/223705671-81a3e2d9-3062-4f56-a6be-7f8d137ac47d.webm)

Since this issue hasn't been updated, and I was on the vacation, I'm not up-to-date on the recent developments. Tell me if I can help you with anything. :relaxed:

**Comment by thrashy** [comment]
Hey! I came across the issue on the spotless repo https://github.com/diffplug/spotless/issues/951 because i was having the same issue. I noticed it wasn't resolved in your repo so i opened a PR for it. Basically, the issue revolves around the source version being set to 1.7 `<setting id="org.eclipse.jdt.core.compiler.source" value="1.7"/>`. I noticed you're on 21 so i bumped all 3 compiler settings to 21
```
        <setting id="org.eclipse.jdt.core.compiler.codegen.targetPlatform" value="21"/>
        <setting id="org.eclipse.jdt.core.compiler.compliance" value="21"/>
        <setting id="org.eclipse.jdt.core.compiler.source" value="21"/>
```
